### ISSUE TITLE
Silent crash on node 6

### ISSUE THREAD

**Issue by scottnonnenberg** [stevenvachon/broken-link-checker#31]
Thanks for this package. So useful! :0)

I downloaded it and tried it on my blog today, and it was giving me just four lines of output. Confused, I started looking at the code, and realized that it finished in the middle of enqueuing its first set of links. With a new `try/catch` in `HtmlChecker/enqueueLink` I found that the process was indeed silently crashing:

```
TypeError: source.hasOwnProperty is not a function
    at cloneObject (lib/internal/linkObj.js:239:14)
    at cloneObject (lib/internal/linkObj.js:245:18)
    at cloneObject (lib/internal/linkObj.js:245:18)
    at Function.linkObj.resolve (lib/internal/linkObj.js:168:64)
    at enqueueLink (lib/public/HtmlChecker.js:158:10)
    at lib/public/HtmlChecker.js:110:5
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

On node `6.1.0` it throws this error, and on node `4.4.3` it works properly. Seems that there are two bugs here - one about the `hasOwnProperty` issue, and the other about silently crashing!

**Comment by scottnonnenberg** [comment]
Figured it out - it's because the value returned by `querystring.parse()` is now created via `Object.create(null)` instead of inheriting from object: https://github.com/nodejs/node/wiki/Breaking-changes-between-v5-and-v6#querystring

Fixing this will also require changes in the `urlobj` repo: https://github.com/stevenvachon/urlobj

Researching now, may have a couple pull requests ready soon...

**Comment by scottnonnenberg** [comment]
I was able to fix the silent error thing by installing `bluebird` as the replacement for `global.Promise` - it provides a key method: `onPossiblyUnhandledRejection()`. In `bin/blc`:

```javascript
var Bluebird = require('bluebird');
global.Promise = Bluebird;

Bluebird.onPossiblyUnhandledRejection(function(error){
  throw error;
});
```

Worth thinking about. Sadly, this method is not available in the [_standard_ promise implementation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

Either that, or get really rigorous always returning rejected promises up the stack. [Lines 97-118 in `HtmlChecker.js`](https://github.com/stevenvachon/broken-link-checker/blob/d6db92731e4fd103ae345dd16baf6fc9247d003b/lib/public/HtmlChecker.js#L97-L118) are a good example: a `call().then()` with no catch and no return. Gotta either return the promise up the stack or handle the error right there. Otherwise errors will be swallowed.

Here's something I wrote about promises: https://blog.scottnonnenberg.com/the-trouble-with-promises/
